src/DataLayouts/cuda.jl

```diff
@@ -76,11 +60,12 @@ function knl_fill!(dest, val)
     j = CUDA.threadIdx().y
 
     h = CUDA.blockIdx().x
-    v = CUDA.blockIdx().y
+    v = CUDA.blockDim().y * (CUDA.blockIdx().y - 1) + CUDA.threadIdx().z
```

Comment:
Should this be 
```
v = CUDA.blockDim().z * (CUDA.blockIdx().y - 1) + CUDA.threadIdx().z
```

Reply:
good catch, thanks

Reply:
i'm worried we didn't pick this up in the tests though...